lib/rspec/rails/feature_check.rb

```diff
@@ -43,8 +43,11 @@ def has_action_mailbox?
         defined?(::ActionMailbox)
       end
 
-      def ruby_3_1?
-        RUBY_VERSION >= "3.1"
+      # TODO: add a self-explanatory method name. See https://github.com/rspec/rspec-rails/pull/2552#issuecomment-1009508693 for hints
```

Comment:
I couldn't. Can you please help me? Is there some explanation, like "the last hash argument is now treated as a positional argument instead of being auto-expanded into kwargs"?
I searched in https://rubyreferences.github.io/rubychanges/3.1.html and couldn't find anything related.

Reply:
So I'm not sure there's a simple description other than "this is how it works for this combo".  The Argument processing in the Rails codebase is pretty complex.  There's an intersection of several changes happening at once, combined with an goal to keep support working from Ruby 2.5 to Ruby 3.1.  That's a lot of moving parts, and I didn't try and dissect exactly what causes the change on input to rspec-rails.

Then rspec-rails is post-processing the arguments with its own conditions, because of how it processes arguments to the matchers.  So making that all line up is complicated, and I basically looked at the discrepancies and found the simplest resolution for the failing case.